test/core/xds/xds_cluster_resource_type_test.cc

```diff
@@ -972,6 +972,40 @@ TEST_F(TlsConfigTest, SystemRootCerts) {
           .certificate_validation_context.ca_certs));
 }
 
+TEST_F(TlsConfigTest, UpstreamTlsContextSni) {
+  ScopedExperimentalEnvVar env_var("GRPC_EXPERIMENTAL_XDS_SNI");
```

Comment:
Please also add a test that we do *not* look at the new fields if the env var is not enabled.  That test can have a TODO indicating that we can remove it when the env var guard goes away.